pandas/tests/io/parser/test_dialect.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+"""
+Tests that dialects are properly handled during parsing
+for all of the parsers defined in parsers.py
+"""
+
+import csv
+
+import pytest
+
+from pandas.compat import StringIO
+from pandas.errors import ParserWarning
+
+from pandas import DataFrame
+import pandas.util.testing as tm
+
+
+def test_dialect(all_parsers):
+    parser = all_parsers
+    data = """\
+label1,label2,label3
+index1,"a,c,e
+index2,b,d,f
+"""
+
+    dia = csv.excel()
+    dia.quoting = csv.QUOTE_NONE
+
+    # Conflicting dialect quoting.
+    with tm.assert_produces_warning(ParserWarning):
+        df = parser.read_csv(StringIO(data), dialect=dia)
```

Comment:
This warning was in the original `dialect.py` and seems odd to me.  Why do we need to explicitly pass in the `quoting` value to be consistent with the dialect's value in order to avoid the warning?  IMO we should only warn if the user passes in a non-default value that conflicts with that of the dialect.

Can do in a follow-up if this makes sense to people.

Reply:
vaguely remember this. yes a followup would be good :>